### DISCUSSION TITLE
Tasmota lacks support for 802.11w (Protected Management Frames)

### DISCUSSION THREAD

**Discussion by Giga-Pudding** [arendst/Tasmota#20888] [Q&A] [answered]
### PROBLEM DESCRIPTION
Tasmota does not support 802.11w (Protected Management Frames). This prevents Tasmota devices to connect to secured wireless networks, in which Protected Management Frames are mandatory. As far as i know, 802.11w is a software feature which is hardware independent.

More informations here:

https://www.wi-fi.org/beacon/philipp-ebbecke/protected-management-frames-enhance-wi-fi-network-security

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [ ] Tasmota binary firmware version number used: _____
  - [ ] Pre-compiled
  - [ ] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
```lua
  Configuration output here:


```
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
```lua
  Rules output here:


```
- [ ] Provide the output of this command: `Status 0`:
```lua
  STATUS 0 output here:


```
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:
```lua
  Console output here:


```

### TO REPRODUCE
Try to connect a Tasmota device to a wireless lan, which is enforcing usage of Protected Management Frames (802.11w).
For example, this is supported by OpenWrt, since version 18.x and newer iirc.

### EXPECTED BEHAVIOUR
-

### SCREENSHOTS
-

### ADDITIONAL CONTEXT
-

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by Jason2866** [comment]
Moved to discussion, since relevant Infos missing for an issue.
What is the concrete issue?

**Reply by sfromis** [reply]
To me, it looks like a feature request.

**Reply by Jason2866** [reply]
Feature Request for what? Use case? What issue will be solved in Tasmota by implementing?

**Reply by Giga-Pudding** [reply]
Sorry folks, I considered this to be an issue, because Tasmota is not able to connect to 802.11w secured WiFi networks. Personally, I think Protected Management Frames are necessary nowadays, especially to prevent denial of service (De-auth Attacks).

Maybe this is more important for me than for others. Please consider this topic, the implementation of 802.11w, as feature request.

**Reply by Jason2866** [reply]
Tasmota supports connecting to WPA3

**Reply by Giga-Pudding** [reply]
Protected Management Frames are a mandatory part of WPA3. But that does not mean, that Protected Management Frames are only available in conjunction with WPA3. You can also have Protected Management Frames (802.11w) with WPA2.

Since [there will be no support of WPA3 for ESP8266 Tasmota devices](https://github.com/arendst/Tasmota/discussions/13414#discussioncomment-1508009), I would like to request at least 802.11w support for ESP8266 devices.

**Reply by Giga-Pudding** [reply]
Okay, i understand that support of 802.11w for ESP8266 devices does not really depend on the Tasmota development team, but on the used ESP8266 SDK.

I learned, that there a different SDKs for the ESP8266:

ESP8266 NonOS SDK (has no support for WPA3 and Protected Management Frames)
ESP8266 RTOS SDK (has support for WPA3 and Protected Management Frames)

Which one is used by Tasmota?

It seems unlikely, that PMF will be available for ESP8266 without WPA3.

**Reply by Jason2866** [reply, marked as answer]
Talking only about esp32. As you correctly wrote there is no WPA3 for esp8266 since NonOS SDK is used as base for esp8266 Arduino. The time of the esp8266 is over, no new features.
